src/main/java/uk/gov/pay/api/json/RequestJsonParser.java

```diff
@@ -70,9 +78,22 @@ static CreatePaymentRequest parsePaymentRequest(JsonNode paymentRequest) {
             createPaymentRequestBuilder.returnUrl(returnUrl);
         }
         
-        if (paymentRequest.has("metadata")) {
-            Map<String, Object> metadata = objectMapper.convertValue(paymentRequest.get("metadata"), Map.class);
-            createPaymentRequestBuilder.metadata(new ExternalMetadata(metadata));
+        if (paymentRequest.has(METADATA)) {
+            Map<String, Object> metadataMap = null;
+            try {
+                metadataMap = objectMapper.convertValue(paymentRequest.get("metadata"), Map.class);
+            } catch (IllegalArgumentException e) {
+                PaymentError paymentError = new PaymentError(METADATA, CREATE_PAYMENT_METADATA_VALIDATION_ERROR, "Field [metadata] must be an object of JSON key-value pairs");
+                throw new WebApplicationException(Response.status(HttpStatus.SC_UNPROCESSABLE_ENTITY).entity(paymentError).build());
+            }
+            ExternalMetadata metadata = new ExternalMetadata(metadataMap);
+            Set<ConstraintViolation<ExternalMetadata>> violations = validator.validate(metadata);
```

Comment:
Seems like a sensible way of doing it which is a stepping stone towards using the validator throughout 👍